File: Examples/AspNetMvcCS/Controllers/FileUploaderController.Overview.cs
using System.Linq;
using System.Web.Mvc;
using GleamTech.FileUltimate.AspNet;
using GleamTech.FileUltimate.AspNet.UI;

namespace GleamTech.FileUltimateExamples.AspNetMvcCS.Controllers
{
    public partial class FileUploaderController
    {
        public ActionResult Overview()
        {
            var fileUploader = new FileUploader
            {
                Width = 600, //Default unit is pixels. Percentage can be specified as CssLength.Percentage(100)
                Height = 300,
                Resizable = true,
                UploadLocation = "~/App_Data/Uploads"
            };

            HandleLanguage(fileUploader);

            return View(fileUploader);
        }

        private void HandleLanguage(FileUploader fileUploader)
        {
            var selectedLanguage = Request["languageSelector"];

            if (selectedLanguage != null)
                fileUploader.DisplayLanguage = selectedLanguage;
            else
                selectedLanguage = fileUploader.DisplayLanguage;

            PopulateLanguageSelector(selectedLanguage);
        }

        private void PopulateLanguageSelector(string selectedLanguage)
        {
            ViewBag.LanguageList = new SelectList(
                FileUltimateWebConfiguration.AvailableDisplayCultures.Select(culture => new
                {
                    Value = culture.Name,
                    Text = culture.NativeName + $" ({culture.Name})"
                }),
                "Value",
                "Text",
                selectedLanguage
            );
        }
    }
}

File: Examples/AspNetMvcVB/Controllers/FileUploaderController.Overview.vb
Imports System.Linq
Imports GleamTech.FileUltimate.AspNet
Imports GleamTech.FileUltimate.AspNet.UI

Namespace Controllers
    Partial Public Class FileUploaderController

        Public Function Overview() As ActionResult
            'Default unit is pixels. Percentage can be specified as CssLength.Percentage(100)
            Dim fileUploader = New FileUploader() With {
             .Width = 600,
             .Height = 300,
             .Resizable = True,
             .UploadLocation = "~/App_Data/Uploads"
            }

            HandleLanguage(fileUploader)

            Return View(fileUploader)
        End Function

        Private Sub HandleLanguage(fileUploader As FileUploader)
            Dim selectedLanguage = Request("languageSelector")

            If selectedLanguage IsNot Nothing Then
                fileUploader.DisplayLanguage = selectedLanguage
            Else
                selectedLanguage = fileUploader.DisplayLanguage
            End If

            PopulateLanguageSelector(selectedLanguage)
        End Sub

        Private Sub PopulateLanguageSelector(selectedLanguage As String)
            ViewBag.LanguageList =  New SelectList(
                FileUltimateWebConfiguration.AvailableDisplayCultures.Select(function(culture) New With {
                        .Value = culture.Name,
                        .Text = culture.NativeName + $" ({culture.Name})"
                    }), 
                "Value",
                "Text",
                selectedLanguage
            )            
        End Sub
    End Class
End Namespace
